devices/Scd30/Scd30Sensor.cs

```diff
@@ -0,0 +1,419 @@
+        ///     previously set reference values from ASC or FRC as well as temperature offset values last setting.
+        /// </para>
+        /// <para>
+        ///     The sensor is able to receive the command at any time, regardless of its internal state.
+        /// </para>
+        /// </summary>
+        public void SoftReset()
+        {
+            ModbusWriteSingleHoldingRegister(0x34, 1);
+        }
+
+        private void ModbusWriteSingleHoldingRegister(ushort register, ushort value)
+        {
+            // Function 0x6 request is always 8 bytes
+            var request = new byte[8];
+            request[0] = Scd30ModbusAddress;
+            request[1] = 0x6;
+            BinaryPrimitives.WriteUInt16BigEndian(new SpanByte(request, 2, 2), register);
+            BinaryPrimitives.WriteUInt16BigEndian(new SpanByte(request, 4, 2), value);
+            BinaryPrimitives.WriteUInt16BigEndian(new SpanByte(request, 6, 2), CalculateModbusRtuCrc16(request, 0, 6));
+
+            // Function 0x6 response is always 8 bytes
+            var response = new byte[8];
+
+            // Execute
+            ModbusWriteAndRead(request, response);
+        }
+
+        private SpanByte ModbusReadHoldingRegisters(ushort register, byte number)
+        {
+            // Function 0x3 request is always 8 bytes
+            var request = new byte[8];
+            request[0] = Scd30ModbusAddress;
+            request[1] = 0x3;
+            BinaryPrimitives.WriteUInt16BigEndian(new SpanByte(request, 2, 2), register);
+            BinaryPrimitives.WriteUInt16BigEndian(new SpanByte(request, 4, 2), number);
+            BinaryPrimitives.WriteUInt16BigEndian(new SpanByte(request, 6, 2), CalculateModbusRtuCrc16(request, 0, 6));
+
+            // Response size depends on the number of registers requested
+            var response = new byte[5 + (2 * number)];
+
+            // Execute
+            ModbusWriteAndRead(request, response);
+
+            // Return data from response
+            return new SpanByte(response, 3, number * 2);
+        }
+
+        private void ModbusWriteAndRead(byte[] request, byte[] response)
+        {
+            // Clear the serial buffer if there's anything
+            while (_serial.BytesToRead > 0)
+            {
+                _serial.ReadByte();
+            }
+
+            // Write the request
+            _serial.Write(request, 0, request.Length);
+
+            // Wait for the expected amount of data (to fill the response array)
+            _serial.WaitForData(response.Length);
+            _serial.Read(response, 0, response.Length);
+
+            // Verify the received message
+            var calculatedCrc = CalculateModbusRtuCrc16(response, 0, response.Length - 2);
+            var receivedCrc = BinaryPrimitives.ReadUInt16BigEndian(new SpanByte(response, response.Length - 2, 2));
```

Comment:
here it's ok to be like this because of the byte array needed for the serial port.